### ISSUE TITLE
Access to low-level requester

### ISSUE THREAD

**Issue by Vampire** [kohsuke/github-api#513]
For being able to use missing API functionality it would be nice if the `Requester` would be accessible, so that the low-level requests can be done manually or e. g. with Kotlin extension functions, but still having the lib do rate limiting and so on.

**Comment by bitwiseman** [comment]
@Vampire 
Seems easy enough (`public` in a few places).  I'm a little concerned though, as then then we'd be committed to the API of the Requester and supporting it.

**Comment by ranma2913** [comment]
This can be done in Groovy already, but isn't possible in Java. Probably not in Kotlin either:
![image](https://github.com/user-attachments/assets/44faf24e-51e9-4a49-b5b6-8a6d67589b20)


```groovy
GHRepository deleteSecret(GHRepository repo, String key) {
    // must use the raw request because kohsuke doesn't have a secret delete yet.
    repo.getRoot()
        .createRequest() // thank you Groovy for allowing this.
        .method("DELETE")
        .withUrlPath("/repos/${repo.getFullName()}/actions/secrets/${key}")
        .send()
    return repo
  }
```

**Comment by Vampire** [comment]
Of course, with Groovy you can even access private stuff easily. And you can do it in Java or Kotlin too with using reflection. But whether you should do it, is a different question, and a stable public API would be preferable

**Comment by bitwiseman** [comment]
@stianst said:

> GitHub is continuously adding new endpoints and fields to existing endpoints, while understandably this library is not always able to support all the new things quickly. The official GitHub CLI client has the same problem, but supports sending any requests using `gh api`.
> 
> It would be nice to be able to create new API requests to endpoints not supported directly by the library, something like `gitHub.createApiRequest`.
> 
> Additionally, for existing endpoints that are supported there is sometimes request parameters or response fields that are not supported. It would be nice to have a way to leverage these without waiting for this library to support them. For example:
> 
> ```
> repository.searchPullRequests().parameter("newParam", "newValue");
> issue.getField(String newField);
> ```
